File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/items/LivingObjectMessageRequestMessage.scala





















// Generated on 05/03/2013 10:34:57
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait LivingObjectMessageRequestMessage extends MessageDeserializer {
  val messageId = 6066

  protected trait Message extends BaseMessage with Serializable {
    val messageId = LivingObjectMessageRequestMessage.messageId

    val msgId: Short
    val parameters: Seq[String]
    val livingObject: UInt

    def serialize(buf: Buffer) {
      buf.writeShort(msgId)
      buf.writeUShort(parameters.length)
      for (entry <- parameters) {
        buf.writeString(entry)
      }
      buf.writeUInt(livingObject)
    }

  }

  protected def create(msgId: Short, parameters: Seq[String], livingObject: UInt): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val msgId = buf.readShort()
    val parameters = {
      val builder = collection.immutable.Seq.newBuilder[String]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readString()
      }
      builder.result()
    }
    val livingObject = buf.readUInt()

    create(msgId, parameters, livingObject)
  }

}

object LivingObjectMessageRequestMessage extends LivingObjectMessageRequestMessage {
  case class MessageImpl(msgId: Short, parameters: Seq[String], livingObject: UInt) extends Message
  type Target = MessageImpl

  override def create(msgId: Short, parameters: Seq[String], livingObject: UInt) = MessageImpl(msgId, parameters, livingObject)

  def apply(msgId: Short, parameters: Seq[String], livingObject: UInt) = create(msgId, parameters, livingObject)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/LivingObjectMessageRequestMessage.java


// Generated on 05/08/2013 19:37:56
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class LivingObjectMessageRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6066;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short msgId;
    public String[] parameters;
    public long livingObject;
    
    public LivingObjectMessageRequestMessage() { }
    
    public LivingObjectMessageRequestMessage(short msgId, String[] parameters, long livingObject) {
        this.msgId = msgId;
        this.parameters = parameters;
        this.livingObject = livingObject;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(msgId);
        buf.writeUShort(parameters.length);
        for (String entry : parameters) {
            buf.writeString(entry);
        }
        buf.writeUInt(livingObject);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        msgId = buf.readShort();
        if (msgId < 0)
            throw new RuntimeException("Forbidden value on msgId = " + msgId + ", it doesn't respect the following condition : msgId < 0");
        int limit = buf.readUShort();
        parameters = new String[limit];
        for (int i = 0; i < limit; i++) {
            parameters[i] = buf.readString();
        }
        livingObject = buf.readUInt();
        if (livingObject < 0 || livingObject > 4294967295L)
            throw new RuntimeException("Forbidden value on livingObject = " + livingObject + ", it doesn't respect the following condition : livingObject < 0 || livingObject > 4294967295L");
    }
    
}
